src/Integration/Suppression/SuppressedIssuesFileStorage.cs

```diff
@@ -0,0 +1,113 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    internal class SuppressedIssuesFileStorage : ISuppressedIssuesFileStorage
+    {
+        private readonly string fileDirectory;
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+
+        public SuppressedIssuesFileStorage(ILogger logger) : this(logger, new FileSystem())
+        {
+            
+        }
+
+        internal SuppressedIssuesFileStorage(ILogger logger, IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+
+            fileDirectory = Path.Combine(Path.GetTempPath(), "SLVS", "Roslyn");
+        }
+
+        public IEnumerable<SonarQubeIssue> Get(string sonarProjectKey)
+        {
+            try
+            {
+                if (!ValidateFileName(sonarProjectKey))
+                {
+                    return Enumerable.Empty<SonarQubeIssue>();
+                }
+
+                var filePath = GetFilePath(sonarProjectKey);
+                var fileContent = fileSystem.File.ReadAllText(filePath);
+                return JsonConvert.DeserializeObject<IEnumerable<SonarQubeIssue>>(fileContent);
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLine($"SuppressedIssuesFileStorage Get Error: {ex.Message}");
+            }
+
+            return Enumerable.Empty<SonarQubeIssue>();
+        }
+
+        public void Update(string sonarProjectKey, IEnumerable<SonarQubeIssue> allSuppressedIssues)
+        {
+            try
+            {
+                if (!ValidateFileName(sonarProjectKey))
+                {
+                    return;
+                }
+
+                var filePath = GetFilePath(sonarProjectKey);
+                var fileContent = JsonConvert.SerializeObject(allSuppressedIssues);
+                fileSystem.Directory.CreateDirectory(fileDirectory);
+                fileSystem.File.WriteAllText(filePath, fileContent);
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLine($"SuppressedIssuesFileStorage Update Error: {ex.Message}");
+            }
+        }
+
+        private bool ValidateFileName(string sonarProjectKey)
+        {
+            if (string.IsNullOrWhiteSpace(sonarProjectKey))
```

Comment:
When would it be valid to call this method with a null/empty string? Wouldn't it be better to throw/assert here?

Reply:
It's because we have talked about not showing the exceptions to users.

Reply:
If we call this method with a null/empty, then it means there is a problem with the our calling code.
We shouldn't silently handle it in that case - we want the failure to be obvious to us during development so we can fix it.

We need to handle runtime errors so the process doesn't crash e.g. some find of I/O or locking failure.
However, we shouldn't be coding defensively to hide logic errors in our code.